lib/avromatic/patches/schema_validator_patch.rb

```diff
@@ -0,0 +1,16 @@
+module Avromatic
+  module Patches
+    module SchemaValidatorPatch
+      # This method replaces validate_recursive in AvroPatches::LogicalTypes::SchemaValidatorPatch
+      # to enable validating datums that contain an encoding provider.
+      def validate_recursive(expected_schema, logical_datum, path, result, encoded = false)
+        datum = resolve_datum(expected_schema, logical_datum, encoded)
+        if datum.is_a?(Hash) && datum.key?(Avromatic::IO::ENCODING_PROVIDER)
+          return if expected_schema.sha256_resolution_fingerprint ==
```

Comment:
How should we handle the `else` case here? Presumably we don't want to call `super` with a hash containing `Avromatic::IO::ENCODING_PROVIDER`.

Reply:
The call to super would result in a validation failure anyway which is what we want but I'll raise an error here to fail faster. The error messages aren't meaningful in our case because Avro::Schema.validate traps errors and returns true/false.